translatorbot/daily-llm.py

```diff
@@ -0,0 +1,219 @@
+import argparse
+import os
+import random
+import re
+import time
+from threading import Thread
+
+from daily import EventHandler, CallClient, Daily
+from datetime import datetime
+from dotenv import load_dotenv
+
+import config
+
+from orchestrator import Orchestrator
+from scenes.story_intro_scene import StoryIntroScene
+from scenes.start_listening_scene import StartListeningScene
+from auth import get_meeting_token, get_room_name
+
+load_dotenv()
+
+class DailyLLM(EventHandler):
+    def __init__(
+            self,
+            room_url=os.getenv("DAILY_URL"),
+            token=os.getenv('DAILY_TOKEN'),
+            bot_name="Translator",
+            language="french", # keep the generated images to a certain theme, like "golden age of illustration," "talented kid's crayon drawing", etc.
+        ):
+
+        # room + bot details
+        self.room_url = room_url
+        self.language = language
+        room_name = get_room_name(room_url)
+        if token:
+            self.token = token
+        else:
+            self.token = get_meeting_token(room_name, os.getenv("DAILY_API_KEY"))
+        # hard-coding this for now so clients can find translators
+        self.bot_name = f"tb-{self.language}"
+
+        duration = os.getenv("BOT_MAX_DURATION")
+        if not duration:
+            duration = 300
+        else:
+            duration = int(duration)
+        self.expiration = time.time() + duration
+        self.story_started = False
+
+        self.finished_talking_at = None
+
+
+        print(f"{room_url} Joining", self.room_url, "as", self.bot_name, "leaving at", self.expiration, "current time is", time.time())
+
+        self.print_debug(f"expiration: {datetime.utcfromtimestamp(self.expiration).strftime('%Y-%m-%d %H:%M:%S')}")
+        self.print_debug(f"now: {datetime.utcfromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S')}")
+
+        self.my_participant_id = None
+
+        self.print_debug("configuring services")
+        self.configure_ai_services()
+        self.print_debug("configuring daily")
+        self.configure_daily()
+
+        self.stop_threads = False
+        self.image = None
+
+        self.print_debug("starting camera thread")
+        self.camera_thread = Thread(target=self.run_camera)
```

Comment:
Disclaimer: I HAVE NOT YET RUN THE DEMO so maybe this question should be self-explanatory. But does the translator contain some kind of camera feed?? I thought it might just be an invisible bot working in the backend.